### ISSUE TITLE
vtexplain allows using a shardmap representing a sharded keyspace with an unsharded vschema;  giving strange results.

### ISSUE THREAD

**Issue by aquarapid** [vitessio/vitess#7675]
schema.sql:
```
CREATE TABLE t1 (
  c1 bigint(20) unsigned NOT NULL,
  c2 varchar(255) NOT NULL,
  PRIMARY KEY (c1)
) ENGINE=InnoDB;
```

vschema.json:
```
{
  "test": {
    "vindexes": {
      "hash": {
        "type": "hash"
      }
    },
    "tables": {
      "t1": {
        "columnVindexes": [
          {
            "column": "t1",
            "name": "hash"
          }
        ]
      }
    }
  }
}
```

(note that the vschema does *not* have a `sharded` directive, so it defaults to `false`).

shards.json (shard map):
```
 {
  "ks1": {
    "-80": {
      "master_alias": {
        "cell": "zone1",
        "uid": 100
      },
      "key_range": {
        "end": "gA=="
      },
      "is_master_serving": true
    },
    "80-": {
      "master_alias": {
        "cell": "zone1",
        "uid": 101
      },
      "key_range": {
        "start": "gA=="
      },
      "is_master_serving": true
    }
  }
}
```

Has 2 shards.

Now, the issue:
  * If you run vtexplain, sometimes a point query would be directed at one shard, sometimes the other, e.g.:
  ```
$ vtexplain -vschema-file ./vschema.json -schema-file ./schema.sql -ks-shard-map-file ./shards.json -sql 'select * from t1 where c1=433'
----------------------------------------------------------------------
select * from t1 where c1=433

1 ks1/-80: select * from t1 where c1 = 433 limit 10001

----------------------------------------------------------------------

Next run:
$ vtexplain -vschema-file ./vschema.json -schema-file ./schema.sql -ks-shard-map-file ./shards.json -sql 'select * from t1 where c1=433'
----------------------------------------------------------------------
select * from t1 where c1=433

1 ks1/80-: select * from t1 where c1 = 433 limit 10001

----------------------------------------------------------------------
```

Using the `-shards` parameter instead of the shard map does not result in this problem.  vtexplain seems to realize that the keyspace is unsharded, and just ignores the shard parameter:

```
$ vtexplain -vschema-file ./vschema.json -schema-file ./schema.sql -shards 2 -sql 'select * from t1 where c1=433'
----------------------------------------------------------------------
select * from t1 where c1=433

1 ks1/-: select * from t1 where c1 = 433 limit 10001

----------------------------------------------------------------------
```

(Note the target shard of `-`)

Version used was recent master bb798e3 ; but older versions like 9.0 are affected as well.

Long story short, we should probably just print a warning if you present a sharded shard map, but do not have a sharded keyspace.
